cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -714,7 +715,81 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
                 AbstractVisibilityEditor.prototype.getContext.call(this),
                 {
                     hide_after_due: this.modelVisibility() === 'hide_after_due',
-                    self_paced: this.model.get('self_paced') === true
+                    self_paced: course.get('self_paced') === true
+                }
+            );
+        }
+    });
+
+    ShowCorrectnessEditor = AbstractEditor.extend({
+        templateName: 'show-correctness-editor',
+        className: 'edit-show-correctness',
+        events: {
+            'change input[name=show-correctness]': 'togglePastDueWarning'
+        },
+
+        modelValue: function() {
+            var showCorrectness = this.model.get('show_correctness');
+            if (showCorrectness === 'always' ||
+                showCorrectness === 'never' ||
+                showCorrectness === 'past_due') {
+                return showCorrectness;
+            }
+            return 'always';
+        },
+
+        afterRender: function() {
+            AbstractEditor.prototype.afterRender.call(this);
+            this.setValue(this.modelValue());
+            this.$('input[name=show-correctness]:checked').change();
+        },
+
+        setValue: function(value) {
+            this.$('input[name=show-correctness][value=' + value + ']').prop('checked', true);
+        },
+
+        currentValue: function() {
+            return this.$('input[name=show-correctness]:checked').val();
+        },
+
+        hasChanges: function() {
+            return this.modelValue() !== this.currentValue();
+        },
+
+        togglePastDueWarning: function() {
+            var warning = this.$('.show-correctness .tip-warning'),
+                display;
+            if (warning) {
+                if (this.currentValue() === 'past_due') {
+                    display = 'block';
+                } else {
+                    display = 'none';
+                }
+                $.each(warning, function(_unused, element) {
+                    $(element).css('display', display);
+                });
+            }
+        },
+
+        getRequestData: function() {
+            if (this.hasChanges()) {
+                return {
+                    publish: 'republish',
+                    metadata: {
+                        show_correctness: this.currentValue()
+                    }
+                };
+            } else {
+                return {};
+            }
+        },
+        getContext: function() {
+            return $.extend(
+                {},
+                AbstractEditor.prototype.getContext.call(this),
+                {
+                    self_paced: course.get('self_paced') === true,
+                    show_correctness_past_due: this.modelValue() === 'past_due'
```

Comment:
Is this the context for the underscore template? I don't see that "show_correctness_past_due" is referenced in the template.